src/views/RecIM/components/Forms/SurfaceForm/index.jsx

```diff
@@ -0,0 +1,93 @@
+import { useState, useMemo } from 'react';
+import Form from '../Form';
+import { createSurface, editSurface } from 'services/recim/match';
+
+const SurfaceForm = ({ surface, closeWithSnackbar, openSurfaceForm, setOpenSurfaceForm }) => {
+  const [errorStatus, setErrorStatus] = useState({
+    Name: false,
+    Description: false,
+  });
+
+  const [isSaving, setSaving] = useState(false);
+
+  const createSurfaceFields = [
+    {
+      label: 'Name',
+      name: 'Name',
+      type: 'text',
+      error: errorStatus.Name,
+      helperText: '*Required',
+      required: true,
+    },
+    {
+      label: 'Description',
+      name: 'Description',
+      type: 'text',
+      error: errorStatus.Name,
+      required: false,
+    },
+  ];
+
+  const currentInfo = useMemo(() => {
+    if (surface) {
+      return {
+        Name: surface.Name,
+        Description: surface.Description,
+      };
+    }
+    return {
+      Name: '',
+      Description: '',
+    };
+  }, [surface]);
+
+  const errorCases = (field, value) => {
+    switch (field) {
+      default:
+        return false;
+    }
+  };
+
+  const handleConfirm = (newInfo, handleWindowClose) => {
+    setSaving(true);
+
+    let surfaceRequest = { ...currentInfo, ...newInfo };
+
+    if (surface) {
+      editSurface(surface.ID, surfaceRequest).then(() => {
+        setSaving(false);
+        closeWithSnackbar({
+          type: 'success',
+          message: 'Surface edited successfully',
+        });
+
+        handleWindowClose();
+      });
+    } else {
+      createSurface(surfaceRequest).then(() => {
+        setSaving(false);
+        closeWithSnackbar({
+          type: 'success',
+          message: 'Surface created successfully',
+        });
```

Comment:
```suggestion
      createSurface(surfaceRequest).then((res) => {
        setSaving(false);
        closeWithSnackbar({
          type: 'success',
          message: `${res.Name} was created successfully`,
        });
```

This is pseudo code but I think we can finally use the objects I'm passing back?

Reply:
It's up to you, it just caught my eye because of the `.then(() => ___)` just having nothing feels weird when I know something gets returned.